agent/agent.go

```diff
@@ -373,6 +370,11 @@ func New(bd BaseDeps) (*Agent, error) {
 	// pass the agent itself so its safe to move here.
 	a.registerCache()
 
+	// TODO: move to newBaseDeps
```

Comment:
Maybe I could have moved this to BaseDeps? I moved it here before rebasing onto master which had the BaseDeps added.

Reply:
I started that work in #8554. It doesn't do the move just yet, but I think it does all the setup to make that possible.

Previously it happened after ` AutoConfig.InitialConfiguration`, but I guess this PR flips the order? I'll read over this now to get a better idea

Reply:
Before, AutoConfig never needed to know about the ACL token so loading after was fine.

Now AutoConfig might actually be performing auto-encrypt in which case it needs the acl token to authorize the request so it has to be loaded earlier. I think this temporary move is probably necessary until #8554 is finished up but it should be a pretty minor merge conflict to resolve.